### ISSUE TITLE
Could not find py4j jar

### ISSUE THREAD

**Issue by thatlittleboy** [autodeployai/pypmml#14]
I'm trying to use pypmml using 

```py
from pypmml import Model
model = Model.fromFile('DecisionTreeIris.pmml')
```

but encountering py4j not found (I've pip-installed it already, and JAVA_HOME is set).

<img width="1079" alt="pm" src="https://user-images.githubusercontent.com/30731072/84226351-1e0bb200-ab14-11ea-93c6-c705cb2ba8d2.png">

How do I debug this? Thanks.

---
(edit)
To add some further context, my environment:

![env](https://user-images.githubusercontent.com/30731072/84226715-26b0b800-ab15-11ea-8174-9a572415ef90.png)

**Comment by scorebot** [comment]
@thatlittleboy From the logs above, I guess the issue could be caused by the py4j included in the pyspark `/usr/lib/spark/python/lib/py4j-0.10.7-src.zip`. Could you check the env `PYTHONPATH` if the py4j of pyspark in the list? If so, remove it, then try again.

**Comment by thatlittleboy** [comment]
Yup, I do see `/usr/lib/spark/python/lib/py4j-0.10.7-src.zip` listed in `echo $PYTHONPATH`.

But unfortunately, I don't think I have permissions to modify environment variables (since it's on JupyterHub hosted by my company).
I don't suppose you see any other way out of this?

**Comment by scorebot** [comment]
@thatlittleboy The `pypmml` uses the default function `find_jar_path` of `java_gateway.py` to find the jar `py4j-0.10.7.jar`, see the code [here](https://github.com/bartdag/py4j/blob/0.10.7/py4j-python/src/py4j/java_gateway.py#L184), so there are several candidate locations to serve the jar, could you try to put a copy to any directory? 
1. `{sys.prefix}/share/py4j`. You need to get the value of sys.prefix of your environment.
2. `/usr/lib/spark/python/lib/py4j-0.10.7-src.zip/py4j/../../../../share/py4j`, that is `/python/lib/share/py4j`

BTW, You can find the jar in the spark distribution: `/usr/lib/spark/jars/py4j-0.10.7.jar`.

**Comment by thatlittleboy** [comment]
It worked!
Actually, more precisely, I couldn't place the jar files into any of the paths searched by `find_jar_path`, since those were all in `/usr/...` and/or a virtualenv folder (`/home/hadoop/miniconda/envs/emr_py36/bin/python`) in which I had no permissions to write into / mkdir.

But your first suggestion to look at `sys.prefix` pointed me in the right direction: I realized I could just change the IPython kernel to one that pointed the sys executable (and thus, the copy of py4j in sitepackages) to one that I did have permissions for. 🙄

Thanks!
